pkg/controllers/jobset_controller_test.go

```diff
@@ -183,6 +192,41 @@ func TestConstructJobsFromTemplate(t *testing.T) {
 					Suspend(false).Obj(),
 			},
 		},
+		{
+			name: "all jobs/pods created with labels and annotations",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				SetAnnotations(annotations).
```

Comment:
I would also like to see SetLabels here also.

Reply:
+1

Reply:
@kannon92 Thanks for catching this.

It looks like we only propagate certain annotations and labels from the top level jobset spec: https://github.com/kubernetes-sigs/jobset/blob/main/pkg/controllers/jobset_controller.go#L767, so setting annotations and labels here actually does nothing.

What is the expected behavior? Should we be setting all labels/annotations from the top level jobset spec on all jobs/pods? Or should I just test for these specific annotations?

Reply:
discussed offline, not all annotations/labels should be propagated

Reply:
I removed the SetAnnotations() from here and instead added different pod/job labels to the test.